_studio/mfx_lib/vpp/src/mfx_vpp_sw_core.cpp

```diff
@@ -1211,17 +1211,21 @@ mfxStatus VideoVPPBase::Reset(mfxVideoParam *par)
         }
     }// Opaque
 
+    bool isCompositionModeEnabledInNewParams = (MFX_ERR_NONE == GetCompositionEnabledStatus(par)) ? true : false;
+
+    // Enabling/disabling composition via Reset() doesn't work currently.
+    // This is a workaround to prevent undefined behavior.
+    if (m_errPrtctState.isCompositionModeEnabled != isCompositionModeEnabledInNewParams)
```

Comment:
Use MFX_CHECK macro instead, it would allow us to see this check in log if MSDK traces would be enabled

Reply:
makes sense. thannks